src/utils/constants.js

```diff
@@ -24,19 +24,24 @@ export const READY_STATE_STAGES = {
 export const MIME_TYPES = {
   MP4: 'video/mp4',
   VND_APPLE_MPEGURL: 'application/vnd.apple.mpegurl',
+  VND_MS_SSTR: 'application/vnd.ms-sstr+xml',
 }
 
 export const MIME_TYPES_BY_EXTENSION = {
   m3u8: MIME_TYPES.VND_APPLE_MPEGURL,
   m3u: MIME_TYPES.VND_APPLE_MPEGURL,
   mp4: MIME_TYPES.MP4,
+  ism: MIME_TYPES.VND_MS_SSTR,
 }
 
 export const getExtension = url => {
   const urlWithoutParameters = url.split('?')[0] //eslint-disable-line
   const match = urlWithoutParameters.match(/\.([a-z0-9]+)$/i)
+  const matchForISM = urlWithoutParameters.match(/[^/].+(\.)(.*?)(?=\/).([a-z0-9]+)$/i)
```

Comment:
According to the test case, we are looking for the pattern `*/*.extension/*`. Couldn't we simplify this regex to something like `.+\/.+\.([a-z0-9]+)\/.+`? It is still somewhat complex, because regexes ... ¯\_(ツ)_/¯, but at least it is more akin to what we are searching for. Plus it does not require the positive lookahead pattern. They are scary!

![image](https://user-images.githubusercontent.com/77497129/124924626-05e5bb00-dfd2-11eb-98c5-07364c5032a4.png)

Besides, I have seen some MSS ism manifests like so: `https://amsv3account-usw22.streaming.media.azure.net/21b17732-0112-4d76-b526-763dcd843449/ignite.ism/manifest(format=fmp4-v20)`. This kind is not identified with the current regex pattern.

Reply:
Nice! My first thought had this approach but your suggestion is so much simpler, indeed. I'll refactor and ping here when push this change (;

Reply:
Done 41de72a